sgv2-dynamo-api/src/main/java/io/stargate/sgv2/dynamosvc/dynamo/QueryProxy.java

```diff
@@ -0,0 +1,243 @@
+    if (opComparisonMatcher.find()) {
+      assert opComparisonMatcher.groupCount() == 5;
+      AttributeValue partitionKeyValue =
+          retrievePartitionKeyValue(
+              opComparisonMatcher.group(1),
+              partitionKey,
+              opComparisonMatcher.group(2),
+              expressionAttributeNames,
+              expressionAttributeValues);
+
+      String sortKeyName = getKeyName(opComparisonMatcher.group(3), expressionAttributeNames);
+      String sortKeyCompOp = opComparisonMatcher.group(4);
+      Predicate predicate = operatorConverter.get(sortKeyCompOp);
+      if (predicate == null) {
+        throw new IllegalArgumentException(sortKeyCompOp + " is not a supported operator");
+      }
+      checkSortKey(sortKeyName, clusteringKey);
+      AttributeValue sortKeyValue =
+          getExpressionAttributeValue(expressionAttributeValues, opComparisonMatcher.group(5));
+      return new QueryBuilder()
+          .select()
+          .from(KEYSPACE_NAME, queryRequest.getTableName())
+          .where(partitionKey.getName(), Predicate.EQ, DataMapper.fromDynamo(partitionKeyValue))
+          .where(sortKeyName, predicate, DataMapper.fromDynamo(sortKeyValue))
+          .build();
+    }
+
+    Matcher betweenMatcher = BETWEEN_PATTERN.matcher(keyConditionExpression);
+    if (betweenMatcher.find()) {
+      assert betweenMatcher.groupCount() == 5;
+      AttributeValue partitionKeyValue =
+          retrievePartitionKeyValue(
+              betweenMatcher.group(1),
+              partitionKey,
+              betweenMatcher.group(2),
+              expressionAttributeNames,
+              expressionAttributeValues);
+
+      String sortKeyName = getKeyName(betweenMatcher.group(3), expressionAttributeNames);
+      checkSortKey(sortKeyName, clusteringKey);
+      AttributeValue lowerBoundValue =
+          getExpressionAttributeValue(expressionAttributeValues, betweenMatcher.group(4));
+      AttributeValue higherBoundValue =
+          getExpressionAttributeValue(expressionAttributeValues, betweenMatcher.group(5));
+      return new QueryBuilder()
+          .select()
+          .from(KEYSPACE_NAME, queryRequest.getTableName())
+          .where(partitionKey.getName(), Predicate.EQ, DataMapper.fromDynamo(partitionKeyValue))
+          .where(sortKeyName, Predicate.GTE, DataMapper.fromDynamo(lowerBoundValue))
+          .where(sortKeyName, Predicate.LTE, DataMapper.fromDynamo(higherBoundValue))
+          .build();
+    }
+
+    Matcher beginWithMatcher = BEGIN_WITH_PATTERN.matcher(keyConditionExpression);
+    if (beginWithMatcher.find()) {
+      // TODO: address this
+      throw new IllegalArgumentException("Cassandra does not support begin_with query");
+    }
+
+    throw new IllegalArgumentException(keyConditionExpression + " is not a valid expression");
+  }
+
+  private String getKeyName(String rawKeyName, Map<String, String> attributeNames) {
+    if (rawKeyName.charAt(0) == '#') {
```

Comment:
May want to document why this is being checked?

Reply:
Sure, I will add some comments. To simply put, a word starting with a pound sign is a placeholder, whose real value needs to be fetched from `ExpressionAttributeNames`.

ref: https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.ExpressionAttributeNames.html